src/builder/PaycallWrapper.sol

```diff
@@ -0,0 +1,28 @@
+// SPDX-License-Identifier: BSD-3-Clause
+pragma solidity ^0.8.23;
+
+import {CodeJarHelper} from "./CodeJarHelper.sol";
+import {IQuarkWallet} from "quark-core/src/interfaces/IQuarkWallet.sol";
+import {Paycall} from "../Paycall.sol";
+
+// Helper library to wrap a QuarkOperation from Actions.sol for a Paycall
+library PaycallWrapper {
+    function wrap(IQuarkWallet.QuarkOperation memory operation, uint256 chainId, uint256 maxPaymentCost)
+        internal
+        pure
+        returns (IQuarkWallet.QuarkOperation memory)
+    {
+        bytes[] memory scriptSources = new bytes[](1);
+        scriptSources[0] = type(Paycall).creationCode;
```

Comment:
This should actually append the `Paycall` script source to the existing `scriptSources` in the `QuarkOperation`.

Reply:
I see. 
So it should be instead...
```
bytes[] memory scriptSources = new bytes[](operation.scriptSources.length + 1);
for(uint i = 0 ; i < operation.scriptSources.length ; i++){
    scriptSources[i] = operation.scriptSources[i];
}
scriptSources[scriptSources.length - 1] = type(Paycall).creationCode;
```
?

Reply:
Yeah that looks correct

Reply:
Fixed  :)